lib/webengine.cpp

```diff
@@ -17,26 +17,60 @@
 Q_GLOBAL_STATIC(SailfishOS::WebEngine, webEngineInstance)
 
 /*!
-  \class SailfishOS::WebEngine
-  \brief Provides access to the Web engine context
-  \inmodule SailfishWebView
+    \class SailfishOS::WebEngine
+    \brief Provides access to the Web engine context.
+    \inmodule SailfishWebView
+    \inherits QMozContext
 
-  Singleton class which provides access to the Web engine context.
+    Singleton class which provides access to the Web engine context.
 */
 
 namespace SailfishOS {
 
 /*!
-  \brief Initialises the WebEngine class.
+    \brief Initialises the WebEngine class.
 
-  Initalises the WebEngine class. The \a profilePath sets the root folder for
-  the mozilla gecko profile. Once set the \a profilePath can't be changed.
+    Initalizes the WebEngine class. The \a profilePath sets the root folder for
+    the mozilla gecko profile. Once set the \a profilePath can't be changed.
+    Set \a runEmbedding to \c false (defaults to true) in order to delay
+    execution of the EmbedLite event loop. In this case, \l runEmbedding must
+    be called manually later.
 
-  Multiple calls to initialize have no effect.
+    Multiple calls to initialize have no effect.
 
-  \sa initialize
+    This method will be called automatically during QML initialisation of the
+    WebView. However under some circumstances it may be useful to perform a
+    custom initalization. In this case \c initialize can be called manually as
+    long as this id done before the QML engine initialization.
+
+    One reason for custom initialization would be to proivde additional
+    component manifests.
+
+    \code
+        SailfishOS::WebEngine::instance() *webEngine = SailfishOS::WebEngine::instance();
+        QString profilePath = QStandardPaths::writableLocation(QStandardPaths::CacheLocation);
+        webEngine->initialize(profilePath, false);
+        webEngine->addComponentManifest(QLatin1String("/path/file.manifest"));
+
+        [...]
+
+        QTimer::singleShot(0, webEngine, SLOT(runEmbedding()));
```

Comment:
... or `QMetaObject::invokeMethod(webEngine, "runEmbedding", Qt::QueuedConnection);`

Reply:
I've taken this directly from [the initialization code](https://github.com/sailfishos/sailfish-components-webview/blob/236086301dd9be6b4089b78278114c65f6069549/lib/webengine.cpp#L70), which I'd prefer to duplicate as it's instructive. Is there a good reason to prefer `invokeMethod` here (in which case, should we change both)?